### ISSUE TITLE
Pemission denied issue

### ISSUE THREAD

**Issue by nmarun** [rajanadar/VaultSharp#138]
Hi Raja,

Thanks for this package to interact with Hashicorp Vault. Here's my snippet where I'm getting a permissions denied issue.

System.AggregateException
  HResult=0x80131500
  Message=One or more errors occurred. ({"errors":["1 error occurred:\n\t* permission denied\n\n"]}
)
  Source=System.Private.CoreLib

string roleId = "roleId1";
string secretId = "secredid1";
string path = "kv/booyah";

IAuthMethodInfo authMethod = new AppRoleAuthMethodInfo(roleId, secretId);
var vaultClientSettings = new VaultClientSettings("https://hcvault.company.com", authMethod)
{
    Namespace = "company/org/team"
};

IVaultClient vaultClient = new VaultClient(vaultClientSettings);

// Use client to read a key-value secret.
Secret<SecretData> kv2Secret = vaultClient.V1.Secrets.KeyValue.V2.ReadSecretAsync(path).Result;

This is how the url structure looks from the vault ui:
ui/vault/secrets/kv/show/booyah?namespace=company%2Forg%2Fteam

Please let me know what I'm missing.

Thanks
Arun

![image](https://user-images.githubusercontent.com/5554175/82807962-887eea00-9ea6-11ea-98af-233113e997e1.png)

**Comment by rajanadar** [comment]
Hi Arun,

Thanks for using the library. Are you sure the vault token generated by the app role (which means the app role 'Role') has permission to read the path that you are trying to read.

Can you please verify the permission using vault CLI or UI but using the vault token generated by the app role. 

The AppRoleAuthInfo gets populated by the vault token when you make the first vault interaction. So you can find the vault token by putting a breakpoint on your exception. You can also find it by using the Beforeapirequestaction delegate or fiddler etc. 

Basically, it seems VaultSharp is generating a vault token, and trying to read the secret. It doesn't seem to have the permissions which sounds like a legit behavior. 

If you can confirm that the same vault token can read the secret using CLI or UI, then it could be a VaultSharp bug. Else, something is wrong with the permissions of the Role.

**Comment by nmarun** [comment]
@rajanadar, thanks for replying back.

I took a second look at it and modified my call to the below:

Secret<SecretData> kv2Secret = vaultClient.V1.Secrets.KeyValue.V2.ReadSecretAsync(path, mountPoint: "kv", version: 1).Result;

This works.

Thanks again.

**Comment by rajanadar** [comment]
Glad to hear
